File: 6kyu/last_survivors_ep2.js
const a = 'abcdefghijklmnopqrstuvwxyza'

function findDups(arr) {
  let dups = [];
  for (let i = 0; i < arr.length; i++) {
      for (let j = 0; j < arr.length; j++)
          if (i !== j)
              if (arr[i] === arr[j]) {
                dups.push(arr[i]);
                break;
              }                                                         
      if (dups.length)
          break;
  }
  return dups;
}

function lastSurvivors(str) {
  let dups = findDups(str.split(''))
  if (dups.length) {
    for (let i = 0; i < dups.length; i++) {
      for (let j = 0; j < 2; j++) {
        let to_remove = str.indexOf(dups[i]);
        str = str.slice(0, to_remove) + str.slice(to_remove + 1, str.length);
      }
	    str += a[a.indexOf(dups[i]) + 1];
	  }
    return lastSurvivors(str);
  }
  return str;
}

File: 6kyu/last_survivors_ep2.py
def last_survivors(s, a='abcdefghijklmnopqrstuvwxyza'):
    s, d = [c for c in s], {k: s.count(k) for k in a[:-1]}
    while any(v > 1 for v in d.values()):
        to_remove = [k for k, v in d.items() if v > 1][0]
        for i in range(2):
            del s[s.index(to_remove)]
        s.append(a[a.index(to_remove) + 1])
        d = {k: s.count(k) for k in a[:-1]}
    return ''.join(s)
